lib/charms/opencti/v0/opencti_connector.py

```diff
@@ -0,0 +1,237 @@
+        """Return the charm directory (the one with charmcraft.yaml in it)."""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.framework.observe(self.on.config_changed, self._reconcile)
+        self.framework.observe(self.on["opencti-connector"].relation_changed, self._reconcile)
+        self.framework.observe(self.on.secret_changed, self._reconcile)
+        self.framework.observe(self.on.upgrade_charm, self._reconcile)
+        self.framework.observe(self.on[self._charm_name].pebble_ready, self._reconcile)
+
+    @property
+    def boolean_style(self) -> str:
+        """Dictate how boolean-typed configurations should translate to environment variable values.
+
+        The style should be either "json" for true/false or "python" for True/False.
+
+        Returns: "json" or "python"
+        """
+        return "json"
+
+    @property
+    def _charm_name(self):
+        """Get charm name.
+
+        Returns:
+            The charm name.
+
+        Raises:
+            RuntimeError: If charm metadata file doesn't exist.
+        """
+        config_file = self.charm_dir / "metadata.yaml"
+        if config_file.exists():
+            return yaml.safe_load(config_file.read_text())["name"]
+        config_file = self.charm_dir / "charmcraft.yaml"
+        if config_file.exists():
+            return yaml.safe_load(config_file.read_text())["name"]
+        raise RuntimeError("charm metadata doesn't exist")
+
+    def _config_metadata(self) -> dict:
+        """Get charm configuration metadata.
+
+        Returns:
+            The charm configuration metadata.
+
+        Raises:
+            RuntimeError: If charm metadata file doesn't exist.
+        """
+        config_file = self.charm_dir / "config.yaml"
+        if config_file.exists():
+            return yaml.safe_load(config_file.read_text())["options"]
+        config_file = self.charm_dir / "charmcraft.yaml"
+        if config_file.exists():
+            return yaml.safe_load(config_file.read_text())["config"]["options"]
+        raise RuntimeError("charm configuration metadata doesn't exist")
+
+    def kebab_to_constant(self, name: str) -> str:
+        """Convert kebab case to constant case
+
+        Args:
+            name: Kebab case name.
+
+        Returns:
+            Input in constant case.
+        """
+        return name.replace("-", "_").upper()
+
+    def _check_config(self) -> None:
+        """Check if required charm configurations are ready.
+
+        Raises:
+            NotReady: If some charm configurations isn't ready.
+        """
+        missing = []
+        for config, config_meta in self._config_metadata().items():
+            value = self.config.get(config)
+            if value is None and not config_meta["description"].strip().startswith("(optional)"):
+                missing.append(config)
+        if missing:
+            raise NotReady("missing configurations: {}".format(", ".join(missing)))
```

Comment:
Perhaps use optional key instead of relying on the description

Reply:
Updated to use the `optional` field.